optimade/models/toplevel.py

```diff
@@ -22,6 +22,8 @@
     "ErrorResponse",
     "EntryInfoResponse",
     "InfoResponse",
+    "AbstractEntryResponseOne",
```

Comment:
I think we should have these not be abstract, and simply be the standard `EntryResponse`s? Since it is needed for provider-specific attributes, and `response_fields` anyway?
(Guessing here that they are "abstract" because `Dict` has been added to the allowed data type?)

Reply:
I would be happy with this, my attempt was just a placeholder. Is there a reason the `StructureResponse`s can also return `Dict`s?

Reply:
Yes, this is needed for the way `response_fields` is handled at the moment, where an amputated `StructureResource` may be supplied to `data` when returning `StructureResponse*`.

Reply:
Ah yes... this does kind of mess up the validator as it can't validate fields that aren't there (duh). I wonder if the best way is for the validator to monkeypatch/emulate the response classes to allow only `StructureResource` to ensure all the the default return fields are correct. Lots of errors being produced for existing implementations, so I can see this being the real use of the tool...

Reply:
If I understand correctly, I agree: The validator should check _complete_ responses, disregarding (in the first run) any parameters (i.e., `filter`, `response_fields`, etc.)